store/etcd/v3/etcd_test.go

```diff
@@ -56,3 +56,26 @@ func TestEtcdV3Store(t *testing.T) {
 	testutils.RunTestTTL(t, kv, ttlKV)
 	testutils.RunCleanup(t, kv)
 }
+
+func TestEtcdListLockKey(t *testing.T) {
+	lockKV := makeEtcdV3Client(t)
+
+	key := "testLocketcdv3"
+	value := []byte("bar")
+
+	// We should be able to create a new lock on key
+	lock, err := lockKV.NewLock(key, &store.LockOptions{Value: value, TTL: 2 * time.Second})
+	assert.NoError(t, err)
+	assert.NotNil(t, lock)
+
+	// Lock should successfully succeed or block
+	lockChan, err := lock.Lock(nil)
+	assert.NoError(t, err)
+	assert.NotNil(t, lockChan)
+
+	pairs, err := lockKV.List(key)
+	assert.NoError(t, err)
+	assert.NotNil(t, pairs)
+	assert.Equal(t, 1, len(pairs))
+	assert.Equal(t, key, pairs[0].Key)
```

Comment:
I think we can remove this test as it is actually supposed to return an empty list with #32.

In place we should check on an non empty list for `testLocketcdv3` with multiple child keys and verify that the resulting list we get from `List` does not have any `_lock` entry.

Reply:
@abronan  Of course, the test has to be modified when #32 will be merged.
I created the two PR separately but when the first one will be merged, I'll do the needed adaptations on the second one :wink: